HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -423,46 +414,50 @@ def self.create_sens_lat_load_actuator_and_equipment(model, name, space, frac_la
     return actuator
   end
 
-  def self.initialize_cfis(model, vent_mech, hvac_map)
+  def self.initialize_cfis(model, mech_vent_fans, hvac_map)
     # Get AirLoop associated with CFIS
-    return if vent_mech.nil? || (vent_mech.fan_type != HPXML::MechVentTypeCFIS)
+    @cfis_airloop = {}
+    @cfis_t_sum_open_var = {}
+    @cfis_f_damper_extra_open_var = {}
+    return if mech_vent_fans.empty?
 
-    cfis_sys_ids = vent_mech.distribution_system.hvac_systems.map { |system| system.id }
+    mech_vent_fans.each do |vent_mech|
+      next unless (vent_mech.fan_type == HPXML::MechVentTypeCFIS)
 
-    @cfis_airloop = nil
+      cfis_sys_ids = vent_mech.distribution_system.hvac_systems.map { |system| system.id }
+      # Get AirLoopHVACs associated with these HVAC systems
+      hvac_map.each do |sys_id, hvacs|
+        next unless cfis_sys_ids.include? sys_id
 
-    # Get AirLoopHVACs associated with these HVAC systems
-    hvac_map.each do |sys_id, hvacs|
-      next unless cfis_sys_ids.include? sys_id
+        hvacs.each do |loop|
+          next unless loop.is_a? OpenStudio::Model::AirLoopHVAC
+          next if (not @cfis_airloop[vent_mech.id].nil?) && (@cfis_airloop[vent_mech.id] == loop) # already assigned
 
-      hvacs.each do |loop|
-        next unless loop.is_a? OpenStudio::Model::AirLoopHVAC
-        next if (not @cfis_airloop.nil?) && (@cfis_airloop == loop) # already assigned
+          fail 'Two airloops found for CFIS.' unless @cfis_airloop[vent_mech.id].nil?
 
-        fail 'Two airloops found for CFIS.' unless @cfis_airloop.nil?
-
-        @cfis_airloop = loop
+          @cfis_airloop[vent_mech.id] = loop
+        end
       end
-    end
 
-    @cfis_t_sum_open_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(' ', '_')}_cfis_t_sum_open") # Sums the time during an hour the CFIS damper has been open
-    @cfis_f_damper_extra_open_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(' ', '_')}_cfis_f_extra_damper_open") # Fraction of timestep the CFIS blower is running while hvac is not operating. Used by infiltration and duct leakage programs
+      @cfis_t_sum_open_var[vent_mech.id] = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(' ', '_')}_cfis_t_sum_open_#{vent_mech.id.gsub(' ', '_')}") # Sums the time during an hour the CFIS damper has been open
+      @cfis_f_damper_extra_open_var[vent_mech.id] = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(' ', '_')}_cfis_f_extra_damper_open_#{vent_mech.id.gsub(' ', '_')}") # Fraction of timestep the CFIS blower is running while hvac is not operating. Used by infiltration and duct leakage programs
```

Comment:
Use indexes instead of ids here too and review rest of code for similar. (These in particular make me more concerned than the previous example because I think EMS has a limit on the number of characters in a line?)

Reply:
We change to use indexes, then maybe these would become arrays instead of hashes?